### ISSUE TITLE
Add support for TimeDelta in time_support

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy#8859]
As discussed in https://github.com/astropy/astropy/pull/8782, we should make sure that ``time_support`` works correctly with ``TimeDelta`` objects. This should be done before the 4.0 release otherwise Matplotlib will treat it as ``Time``, which would be incorrect.

**Comment by mhvk** [comment]
To add to this: `TimeDelta` is of course quite different, much more like a `Quantity`, although with special notation possible (like `Angle`). The easiest first step might be to treat it as a `Quantity` in units of `day` or `second` (depending on its `format`).

**Comment by astrofrog** [comment]
One thing that it might be nice to allow is to still be able to use hh:mm:ss style notation (optionally) for ``TimeDelta``, but then indeed also allow decimal days, seconds, etc.

**Comment by mhvk** [comment]
Agreed, though that will be good for `Angle` and `Quantity` with time units as well...

**Comment by bsipocz** [comment]
Above, and in the previous discussion you both said this needs to be done in 4.0. Is it still on the table?

**Comment by mhvk** [comment]
@astrofrog - my sense would be to just work with the `Quantity` analogy for now; the easiest way to get things like `hh:mm:ss.sss` would be to add a new `TimeDeltaString` format or so (or, perhaps, do things already work with 'datetime' format!?).

Actually, very easy would be to just support `TimeDelta` under `quantity_support` - and/or should we have `astropy_support`?

**Comment by astrofrog** [comment]
@mhvk - I agree that it's difficult to know if this should go in quantity_support or time_support. In fact, regarding the naming of this, having taught a workshop where we used ``quantity_support`` last week the question came up as to why this isn't just the default when astropy.units is imported and I didn't have a good answer for that. For ``time_support``, we need to be able to specify some kwargs, but in fact there's no reason this can't also just be enabled by default and have ``time_support`` be available to help change the default format/scales used. If we did that we could just not have ``quantity_support`` and rename ``time_support`` to ``time_plot_options`` for example.

**Comment by mhvk** [comment]
I guess we didn't do it before because not all matplotlib supported it that well. But your plan sounds eminently reasonable!
